modules/dns/dns_api.rb

```diff
@@ -73,6 +86,18 @@ class Api < ::Sinatra::Base
       end
     end
 
+    def validate_srv_value!(value)
+      priority,weight,port,target = value.split(' ')
```

Comment:
Not an expert on ruby, but you probably need to ensure that it also handles values where it isn't 4. Could you write tests for this function?